Bluetooth: Mesh: remove 20ms tx delay in adv bearer

alxelax opened the pull request:
The recommendation to have 20ms is fair for two
consecutive messages over a single bearer. 
When mesh sends two messages it can be interpreted as two
bearers working in parallel. No need to keep
an artificial 20ms delay for that. Delay was
removed and all related bsim tests were fixed.

PavelVPV commented on subsys/bluetooth/mesh/adv_ext.c:
```diff
@@ -335,19 +325,7 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv)
-		delta = k_uptime_delta(&timestamp);
-		k_work_reschedule(&ext_adv->work, K_MSEC(ADV_INT_FAST_MS - delta));
-	}
+	k_work_submit(&ext_adv->work);
```
Putting comment here as I can't put it at line 266. `schedule_send` at line 266 should be redesigned as it will just run the same `send_pending_adv` work again.

alxelax replied:
No, it shouldn't. The same work will not be run since flag `ADV_FLAG_ACTIVE` has been already cleared after sending. Even more, if work is submitted again while the ongoing work was in the submitted queue, this work will be lost. Probably adv structure will leak over fifo since it is put to fifo but task to get it is not submitted.
Without this, about 30-40% bsim tests do not work.

PavelVPV replied:
By redesign I don't mean you need to not call `schedule_send` in other places. But you can go through these flags checks and if it comes to `k_work_submit` just continue here.

PavelVPV replied:
Something like this: https://github.com/PavelVPV/zephyr/commit/cd923d784ff4a6ea854326fc8df1aa27e3a3286d

alxelax replied:
I'm sure it is not worth to mix the redesign of internal states aka flags handling (very ugly approach comparing to normal fsm) with removing delay task. They are not related to each other. If we do not want replanning work over work, it should be done separately and shouldn't block ongoing changes. 
Removing of delay doesn't change work usage paradigma.

PavelVPV replied:
Please see the commit from the message above. And yes, perhaps flags can also be redesigned.

alxelax replied:
I do not think I agree with suggested changes, since they really submit work twice.
As I said before, refactoring should be done as separate task with fixing of impacted bsim tests.

PavelVPV replied:
> I do not think I agree with suggested changes, since they really submit work twice.

As you may see I removed re-submitting the work from `send_pending_adv`.

> As I said before, refactoring should be done as separate task with fixing of impacted bsim tests.

They shouldn't be impacted really if the behavior of the advertiser stays the same. Only the blob transfer in push mode failed at my desk with the commit above when I ran it in parallel mode, but didn't fail when I ran it separately which is suspicious. If you feel OK with this change, then fine, I'll create a task to fix this later.

PavelVPV replied:
NCSDK-25117

PavelVPV reviewed (commented):
Will these bsim changes be "fixed" when we implement randomization for beacons? If so, could you please move these changes to a separate commit so that we could revert it later?

alxelax commented:
> Will these bsim changes be "fixed" when we implement randomization for beacons? If so, could you please move these changes to a separate commit so that we could revert it later?

It might be fixed, but might be not. I'm pretty sure my changes are correct to fix test scenario in general manner. Test scenario waits for the subnetwork beacon and then sends a beacon back. If a subnetwork is the beginning of the list, the two devices send beacons to meet each other. Test will fall from time to time, that depends on added or removed changes that adds\removes some delays here and there. My changes guarantee that there will not be collisions since rx\tx beacon sending are planned in different time.

PavelVPV commented:
> > Will these bsim changes be "fixed" when we implement randomization for beacons? If so, could you please move these changes to a separate commit so that we could revert it later?
> 
> It might be fixed, but might be not. I'm pretty sure my changes are correct to fix test scenario in general manner. Test scenario waits for the subnetwork beacon and then sends a beacon back. If a subnetwork is at the beginning of the list, the two devices send beacons to meet each other. Test will fall from time to time, that depends on added or removed changes that adds\removes some delays here and there. My changes guarantee that there will not be collisions since rx\tx beacon sending are planned in different time.

Well, it was working before because beacons from multiple subnets were split in time. Now, as per my understanding, in some tests, the dut is not able to receive a beacon from the tester as it doesn't listens to as it sends next beacon (right?), in other tests the receiver is not able to receive beacon, so `BEACON_INTERVAL + ` just increases the probability. Once we add a delay between beacons, the behavior should be the same as it was before this change. If we change the tests, how can we be sure that it works like it worked before?

alxelax commented:
> > > Will these bsim changes be "fixed" when we implement randomization for beacons? If so, could you please move these changes to a separate commit so that we could revert it later?
> > 
> > 
> > It might be fixed, but might be not. I'm pretty sure my changes are correct to fix test scenario in general manner. Test scenario waits for the subnetwork beacon and then sends a beacon back. If a subnetwork is at the beginning of the list, the two devices send beacons to meet each other. Test will fall from time to time, that depends on added or removed changes that adds\removes some delays here and there. My changes guarantee that there will not be collisions since rx\tx beacon sending are planned in different time.
> 
> Well, it was working before because beacons from multiple subnets were split in time. Now, as per my understanding, in some tests, the dut is not able to receive a beacon from the tester as it doesn't listens to as it sends next beacon (right?), in other tests the receiver is not able to receive beacon, so `BEACON_INTERVAL + ` just increases the probability. Once we add a delay between beacons, the behavior should be the same as it was before this change. If we change the tests, how can we be sure that it works like it worked before?

I've already described above the reason of failed beacon tests with multiple subnetwork. Any artificial delays do not guarantee that the test will pass. Probability of failed test will always be > 0. This is the nature of an implemented test scenario.

Everywhere I added `BEACON_INTERVAL + ` I have already fixed since these are not adv beacons. It just relaxes waiting conditions for beacons over gatt.

Ok, I'll make the separate commit if you insist, but I do not see a lot of sense in this. Any artificial beacon delay in between will not return behavior previous back, just add the new one with the new implications and broken bsim tests.

alxelax committed:
tests: Bluetooth: Mesh: relax beacon test time conditions

Some beacon tests have strict time conditions and depend
on randomization interval that doesn't guarantee test pass.
This change fixes strict conditions.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

alxelax committed:
Bluetooth: Mesh: remove 20ms tx delay in adv bearer

The recommendation to have 20ms is fair for two
consecutive messages over a single bearer. When mesh
sends two messages it can be interpreted as two
bearers working in parallel. No need to keep
an artificial 20ms delay for that. Delay was
removed and all related bsim tests were fixed.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>
